### ISSUE TITLE
SSL bad handshake when using certain libraries

### ISSUE THREAD

**Issue by benc-uk** [microsoft/AzureNotebooks#486]
Please describe your issue as completely as possible.

#### Which site did you access us from?
    [⭐️] https://notebooks.azure.com
    [] https://studio.azureml.net

#### Describe the issue you are encountering
When using the Azure Cognitive Services libraries to process a file, the following error is thrown
```
ClientRequestError: Error occurred in request., ConnectionError: ("bad handshake: Error([('SSL routines', 'SSL23_GET_SERVER_HELLO', 'unknown protocol')],)",)
```
This exact same code runs perfectly on my local machine both in Jupyter and as standalone Python code. I think the SSL error is caused by the underlying HTTPS call to the Cognitive Services URL endpoint

#### What steps can we take to reproduce this issue? If you have this in a notebook already, you can just share a link here.
https://notebooks.azure.com/bencoleman/projects/ml-experiments/html/cognitive-services/computer-vision.ipynb

#### Which language/kernel were you using when you experienced this issue?
    [] Python 2.7 
    [] Python 3.5
    [⭐️] Python 3.6 
    [] R 
    [] F#
    [] Terminal

#### What time (please include time zone) did this occur?
18:00 GMT+1

#### Which Web Browser(s) did you see the error in?
    [⭐️] Chrome
    [] Firefox
    [] Internet Explorer / Edge (Please note this is not supported)
    [] Other (Please Specify)

#### Which Operating System are you using? Please specify version as well if possible:
    [⭐️] Windows
    [] MacOS
    [] Ubuntu
    [] Other (Please Specify)

#### Each notebook instance has log files. It may be helpful for us to have them. You can retrieve them by following this [help article](https://notebooks.azure.com/help/jupyter-notebooks/troubleshooting/jupyter-debug-log)
There were no logs `cat: .nb.log: No such file or directory`

**Comment by mdbrannock** [comment]
@benc-uk did you ever have any luck fixing this? I'm running into the same issue.

**Comment by benc-uk** [comment]
Nope, and no workaround. Doesn't seem like anyone looks at the issues here 
I have a feeling it is due to it's the outdated version of OpenSSL installed on the VMs running the Notebook service

**Comment by mdbrannock** [comment]
Lovely. For what it's worth, I was getting the same error but it was resolved in a way that still doesn't make any sense to me. I was trying to pass a binary in a post request as an `octet-stream`. The actual object that I was passing as a binary was of type `_io.BytesIO`, which was apparently not right. When I passed an object of type `bytes` then (for reasons that remain unclear) I stopped getting the Bad Handshake error.

I have no clue if that will help you or any future people, nor do I understand what the relationship is between the type of object passed and any OpenSSL issues, but there it is.
